runtime-k3d/src/gen_helm.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (c) 2023 Robert Bosch GmbH and Microsoft Corporation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import yaml
+from yaml.loader import SafeLoader
+from typing import Optional
+import os
+
+from lib import (
+    get_services,
+    get_script_path,
+    create_nodeport_spec,
+    create_cluster_ip_spec,
+    get_template
+)
+
+node_port = 30050
+
+def find_service_spec_index(lst, name):
+    """Returns index of spec for the given name.
+    Args:
+        lst: The list of the template specifications.
```

Comment:
I would suggest a better name for that variable - maybe template_value? Also adding some type hints would make it easier to understand what that variable is so it doesnt need to have "list" in the name